langchain/document_loaders/etherscan.py

```diff
@@ -0,0 +1,164 @@
+import os
+import re
+from typing import List
+
+import requests
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+
+class EtherscanLoader(BaseLoader):
+    """
+    Load transactions from an account on Ethereum mainnet.
+
+    The Loader use Etherscan API to interact with Ethereum mainnet.
+
+    ETHERSCAN_API_KEY environment variable must be set use this loader.
+    """
+
+    def __init__(
+        self,
+        account_address: str,
+        api_key: str = "docs-demo",
+        filter: str = "normal_transaction",
+        page: int = 1,
+        offset: int = 10,
+        start_block: int = 0,
+        end_block: int = 99999999,
+        sort: str = "desc",
+    ):
+        self.account_address = account_address
+        self.api_key = os.environ.get("ETHERSCAN_API_KEY") or api_key
+        self.filter = filter
+        self.page = page
+        self.offset = offset
+        self.start_block = start_block
+        self.end_block = end_block
+        self.sort = sort
+
+        if not self.api_key:
+            raise ValueError("Etherscan API key not provided")
+
+        if not re.match(r"^0x[a-fA-F0-9]{40}$", self.account_address):
+            raise ValueError(f"Invalid contract address {self.account_address}")
+        if filter not in [
+            "normal_transaction",
+            "internal_transaction",
+            "erc20_transaction",
+            "eth_balance",
+            "erc721_transaction",
+            "erc1155_transaction",
+        ]:
+            raise ValueError(f"Invalid filter {filter}")
+
+    def load(self) -> List[Document]:
```

Comment:
Nice addition!

Nit: make this `lazy_load` and `yield` documents.

Then add as load method:
```
    def load(self) -> List[Document]:
        """Load Documents from table."""
        return list(self.lazy_load())
```

Useful if folks want to lazy load large volumes of data.

See example:
https://github.com/hwchase17/langchain/blob/master/langchain/document_loaders/airtable.py